homeassistant/components/camera/device_condition.py

```diff
@@ -0,0 +1,82 @@
+"""Provide the device conditions for Camera."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_CONDITION,
+    CONF_DEVICE_ID,
+    CONF_DOMAIN,
+    CONF_ENTITY_ID,
+    CONF_TYPE,
+    STATE_IDLE,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import condition, config_validation as cv, entity_registry
+from homeassistant.helpers.config_validation import DEVICE_CONDITION_BASE_SCHEMA
+from homeassistant.helpers.typing import ConfigType, TemplateVarsType
+
+from . import DOMAIN
+
+# TODO specify your supported condition types.
+CONDITION_TYPES = {
+    "is_on",
+    "is_off",
+}
+
+CONDITION_SCHEMA = DEVICE_CONDITION_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
+        vol.Required(CONF_TYPE): vol.In(CONDITION_TYPES),
+    }
+)
+
+
+async def async_get_conditions(
+    hass: HomeAssistant, device_id: str
+) -> list[dict[str, str]]:
+    """List device conditions for Camera devices."""
+    registry = await entity_registry.async_get_registry(hass)
+    conditions = []
+
+    # Get all the integrations entities for this device
+    for entry in entity_registry.async_entries_for_device(registry, device_id):
+        if entry.domain != DOMAIN:
+            continue
+
+        # Add conditions for each entity that belongs to this integration
+        base_condition = {
+            CONF_CONDITION: "device",
+            CONF_DEVICE_ID: device_id,
+            CONF_DOMAIN: DOMAIN,
+            CONF_ENTITY_ID: entry.entity_id,
+        }
+
+        conditions += [{**base_condition, CONF_TYPE: cond} for cond in CONDITION_TYPES]
+
+    return conditions
+
+
+@callback
+def async_condition_from_config(
```

Comment:
I am not positive that the implementation of this method is correct.   Perhaps you can help me see what i'm missing, or confirm that the behavior.

My read of the camera state machine is that the state can either be `recording`, `streaming`, or `idle` and on/off is a separate property:
https://github.com/home-assistant/core/blob/f5e51ad5f6e53c840323b2a81098d9425cb197c2/homeassistant/components/camera/__init__.py#L617

This is also described here: https://developers.home-assistant.io/docs/core/entity/camera/#properties

However, there is a comment that says `_attr_state: None = None  # State is determined by is_on` but i'm not sure that is actually correct.

What do you think?

Reply:
You're right, @allenporter, the camera state is `recording`, `streaming` or `idle`.